### ISSUE TITLE
0.8 - async should produce an unsupported error msg when used with with_items

### ISSUE THREAD

**Issue by cigor** [ansible/ansible#921]
```

- name: test async vs loop
  user: root
  hosts: all
  tasks:
   - name: sleep $item
     action: command sleep $item; echo done
     with_items:
      - 10
      - 15
      - 25
     async: 60
     poll: 2
```

This playbook will fail consistently with the following error message:

ok: [somemachine.loc] => (item=25) => {"ansible_job_id": "944937052083", "item": 25, "module": "command", "results_file": "/somepath/.ansible_async/944937052083", "started": 1}
ERROR: unexpected error: unable to determine jid

It seems all instances for the different items of the command use the same results_file, which gets cleaned up after the first item. finishes.

$ ansible-playbook --version
ansible-playbook 0.7 (devel 1ae018ce94) last updated 2012/08/21 11:04:40 (GMT +200)

**Comment by mpdehaan** [comment]
Yeah, we're not going to fix this, but we should make it produce a reasonable error.   Async also does not make sense with 'copy', 'assemble', or 'template' -- only purely remote modules without server side components.

**Comment by mpdehaan** [comment]
A way to implement this in 0.8 would be to check if there was a plugin named after the module, and if so, disallow async.   Requires merging in my modular actions patch first.

**Comment by mpdehaan** [comment]
Fixed in 8e9344e
